recipe/meta.yaml

```diff
@@ -94,25 +90,27 @@ outputs:
         - arm-variant * {{ arm_variant_type }}  # [aarch64]
         - cuda-version {{ cuda_version }}
         - cuda-nvcc-tools {{ cuda_version }}
+        - cuda-crt-dev_{{ target_platform }}
+        - cuda-nvvm-dev_{{ target_platform }}
```

Comment:
Do these need to use `{{ pin_subpackage(...) }}`, in `host` too? Why/why not?

Reply:
Good question, I'm not really sure on this one and was trying to follow what some other outputs do.

[Here](https://github.com/conda-forge/cuda-nvcc-impl-feedstock/blob/main/recipe/meta.yaml#L96) we have `host` without `pin_subpackage`
And [here](https://github.com/conda-forge/cuda-nvcc-impl-feedstock/blob/main/recipe/meta.yaml#L146) we have `run` with `pin_subpackage`

But looking now in cudart we have `host` and `run` with `pin_subpackage` [here](https://github.com/conda-forge/cuda-cudart-feedstock/blob/main/recipe/meta.yaml#L83-L86)
And also `run` without `pin_subpackage` [here](https://github.com/conda-forge/cuda-cudart-feedstock/blob/main/recipe/meta.yaml#L173)

Reply:
The docs say: https://docs.conda.io/projects/conda-build/en/stable/resources/variants.html#extra-jinja2-functions

> `pin_subpackage('package_name', min_pin='x.x.x.x.x.x', max_pin='x', exact=False)`: To be used as pin in run and/or test requirements. Takes package name argument. Used to refer to particular versions of subpackages built by parent recipe as dependencies elsewhere in that recipe. Can use either pinning expressions, or exact (including build string).

I'm not sure if that is meant to include `build` / `host` pinnings or not (docs don't mention those explicitly). Since we've seen it work correctly for `cuda-cudart`, I would consider adding `pin_subpackage` to `host` here as well. It feels more explicit. @jakirkham please weigh in if you know more.

Reply:
If it is a subpackage in the same recipe, I recommend using pin_subpackage wherever that package is referenced. The one exception is that if the subpackage provides a run_exports, then you would list the pin_subpackage only in the host dependencies, since run_exports would take care of it in run.

Reply:
Yeah let's add it. Think we want something like this. Though please feel free to correct me if I've typoed something

```suggestion
        - cuda-crt-dev_{{ target_platform }}
        - {{ pin_subpackage("cuda-nvvm-dev_" ~ target_platform, exact=True) }}
```

Reply:
Switched these over here  https://github.com/conda-forge/cuda-nvcc-impl-feedstock/pull/8/commits/3f5e527d3dd0eff28876ca70fea75bee6c7c9425